exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/CreateFunctionHandler.java

```diff
@@ -175,22 +175,20 @@ private void initRemoteRegistration(List<String> functions,
     List<Jar> remoteJars = remoteRegistry.getRegistry(version).getJarList();
     validateAgainstRemoteRegistry(remoteJars, jarManager.getBinaryName(), functions);
     jarManager.copyToRegistryArea();
-    boolean cleanUp = true;
     List<Jar> jars = Lists.newArrayList(remoteJars);
     jars.add(Jar.newBuilder().setName(jarManager.getBinaryName()).addAllFunctionSignature(functions).build());
     Registry updatedRegistry = Registry.newBuilder().addAllJar(jars).build();
     try {
       remoteRegistry.updateRegistry(updatedRegistry, version);
-      cleanUp = false;
     } catch (VersionMismatchException ex) {
+      jarManager.deleteQuietlyFromRegistryArea();
       if (retryAttempts-- == 0) {
         throw new DrillRuntimeException("Failed to update remote function registry. Exceeded retry attempts limit.");
       }
       initRemoteRegistration(functions, jarManager, remoteRegistry, retryAttempts);
-    } finally {
-      if (cleanUp) {
-        jarManager.deleteQuietlyFromRegistryArea();
-      }
+    } catch (Exception e) {
```

Comment:
Since recursion is used on retry, won't we end up deleting the jar multiple times? No harm, but seems odd...